app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -19,6 +19,8 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
     DEFAULT_PORT = 80
     default_value_for :port, DEFAULT_PORT
 
+    has_many :middleware_domains, :foreign_key => :ems_id, :dependent => :destroy
+    has_many :middleware_server_groups, :foreign_key => :ems_id, :dependent => :destroy
```

Comment:
> @abonas Looking into containers model and the provider has the reference to all entities https://git.io/vK7Fv

yes, iirc all the relationships exist there because there was no functional requirement not to have them

> If I remove the provider - server group relationship I have troubles in save_inventory_middleware.rb (when calling ems.middleware_server_groups) Do you know about any provider who has an entity that is not "owned" by the provider?

a quick search in the code didn't yield any results, but there are a lot of providers and models and some cases have relationships in other files (modules/mixins), so it's worth to ask regarding it on gitter.

> If the provider - entity relationship is there, we can still hide it in the UI, right?

afaik - yes